tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_HarmonicSchwarzschild.cpp

```diff
@@ -0,0 +1,292 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <utility>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/PointwiseFunctions/AnalyticSolutions/TestHelpers.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/HarmonicSchwarzschild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ExtrinsicCurvature.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma: no_forward_declare Tags::deriv
+
+namespace {
+// Get test coordinates
+template <typename Frame, typename DataType>
+tnsr::I<DataType, 3, Frame> spatial_coords(const DataType& used_for_size) {
+  auto x = make_with_value<tnsr::I<DataType, 3, Frame>>(used_for_size, 0.0);
+  get<0>(x) = 1.32;
+  get<1>(x) = 0.82;
+  get<2>(x) = 1.24;
+  return x;
+}
+
+template <typename Frame, typename DataType>
+void test_tag_retrieval(const DataType& used_for_size) {
+  // Parameters for HarmonicSchwarzschild solution
+  const double mass = 1.234;
+  const std::array<double, 3> center{{1.0, 2.0, 3.0}};
+  const auto x = spatial_coords<Frame>(used_for_size);
+  const double t = 1.3;
+
+  // Evaluate solution
+  const gr::Solutions::HarmonicSchwarzschild solution(mass, center);
+  TestHelpers::AnalyticSolutions::test_tag_retrieval(
+      solution, x, t,
+      typename gr::Solutions::HarmonicSchwarzschild::template tags<DataType,
+                                                                   Frame>{});
+}
+
+void test_serialize() {
+  gr::Solutions::HarmonicSchwarzschild solution(3.0, {{0.0, 3.0, 4.0}});
+  test_serialization(solution);
+}
+
+void test_copy_and_move() {
+  gr::Solutions::HarmonicSchwarzschild solution(3.0, {{0.0, 3.0, 4.0}});
+  test_copy_semantics(solution);
+  auto solution_copy = solution;
+  // clang-tidy: std::move of trivially copyable type
+  test_move_semantics(std::move(solution), solution_copy);  // NOLINT
+}
+
+void test_construct_from_options() {
+  const auto created =
+      TestHelpers::test_creation<gr::Solutions::HarmonicSchwarzschild>(
+          "Mass: 0.5\n"
+          "Center: [1.0,3.0,2.0]");
+  CHECK(created ==
+        gr::Solutions::HarmonicSchwarzschild(0.5, {{1.0, 3.0, 2.0}}));
+}
+
+// Test that computed spacetime quantities are computed as expected. See
```

Comment:
Please add a test that the solution solves the Einstein equations. See `TestHelpers::VerifyGrSolution`. For me that is the most important test. Re-implementations of the equations, as done in this function, have the tendency to have the same bugs as the original implementation ;)

Reply:
I second this suggestion